sqlmesh/core/snapshot/definition.py

```diff
@@ -646,7 +651,18 @@ def missing_intervals(
         )
         latest_ts = to_timestamp(latest)
 
-        croniter = self.model.croniter(start_ts)
+        upper_bound_ts = (
+            to_timestamp(self.model.cron_floor(execution_time or now()))
```

Comment:
so the upper bound is computed with the regular cron expression, and on line 662 we check it against the normalized interval unit cron?

Reply:
That is correct. Do you see a problem here?

Reply:
no